### PR TITLE
virttest.utils_test: Small bug fix for get_time()

### PR DESCRIPTION
**Pull request by spcui**
Signed-off-by: Shuping Cui <scui@redhat.com>

### CODE REVIEW

**Comment by spcui** [comment]
ID: 1234667

**Comment by spcui** [comment]
# ./tools/scan_results.py 
Test                                                                                                                                                                                              Status     Seconds  Info
----                                                                                                                                                                                              ------     -------  ----
        (Result file: /root/tests/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.8192m.repeat1.run_test.Host_RHEL.6.6.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.ntp.with_load.default_load     GOOD       136      completed successfully
virt.qemu.smp_4.8192m.repeat1.run_test.Host_RHEL.6.6.qcow2.virtio_blk.up.virtio_net.RHEL.7.1.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_load.default_load    GOOD       122      completed successfully
----                                                                                                                                                                                              GOOD       264

File: virttest/utils_test/__init__.py [hunk 1]
```diff
@@ -290,10 +290,10 @@ def get_time(session, time_command, time_filter_re, time_format):
         reo = None
 
         try:
-            reo = re.findall(time_filter_re, output)[0]
+            reo = re.findall(time_filter_re, output)
+            str_time = reo[0]
```
Comment (suqinhuang) [inline virttest/utils_test/__init__.py]: Hi scui,
for these two lines. your update is the same as original one

File: virttest/utils_test/__init__.py [hunk 2]
```diff
@@ -304,7 +304,7 @@ def get_time(session, time_command, time_filter_re, time_format):
                           time_filter_re, output)
             raise err
 
-        guest_time = time.mktime(time.strptime(reo, time_format)) + num
+        guest_time = time.mktime(time.strptime(str_time, time_format)) + num
```
Comment (suqinhuang) [inline virttest/utils_test/__init__.py]: utils_test.get_time is used in :
timedrift_with_migration.py 
timedrift_no_net.py 
timedrift_no_net_win.py time_manage.py 
timedrift_with_reboot.py
timedrift_with_stop.py
timerdevice_boot.py 
timedrift_with_cpu_offline.py

suggest to check all the cases if you update it

Comment (spcui) [inline virttest/utils_test/__init__.py]: Hi @suqinhuang tested the above cases, already fixed the error now.

**Comment by suqinhuang** [comment]
Hi scui,
could you provide all the test result here.

Regards,
Suqin

**Comment by spcui** [comment]
Hi @suqinhuang 
# ./tools/scan_results.py 
Test                                                                                                                                                                                              Status     Seconds  Info
----                                                                                                                                                                                              ------     -------  ----
        (Result file: /root/staf-kvm-devel/autotest-devel/client/results/default/status)
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_load.default_load    GOOD       126      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_migration            GOOD       124      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_reboot               GOOD       42       completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_stop.clock_host      GOOD       98       completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.shared_ntp_date.date.with_stop.clock_vm        GOOD       98       completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.monotonicity                                   GOOD       108      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.with_cpu_offline                               GOOD       669      completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.with_signal_stop.clock_host                    GOOD       98       completed successfully
virt.qemu.smp_4.4096m.repeat1.run_test.Host_RHEL.7.2.qcow2.virtio_blk.up.virtio_net.RHEL.7.2.x86_64.io-github-autotest-qemu.timedrift.with_pvclock.with_signal_stop.clock_vm                      GOOD       98       completed successfully
----                                                                                                                                                                                              GOOD       1497

**Comment by PandaWei** [comment]
@spcui 
what's meaning of (ID: 1234667)?
It's better to make some explanation,
attach the error message if it's a bug.

**Comment by spcui** [comment]
@PandaWei 
Yes, it's a bug, error logs:
Traceback (most recent call last):
  File "/home/autotest/autotest-devel/client/shared/test.py", line 823, in _call_test_function
    return func(*args, **dargs)
  File "/home/autotest/autotest-devel/client/shared/test.py", line 291, in execute
    postprocess_profiled_run, args, dargs)
  File "/home/autotest/autotest-devel/client/shared/test.py", line 209, in _call_run_once
    *args, **dargs)
  File "/home/autotest/autotest-devel/client/shared/test.py", line 313, in run_once_profiling
    self.run_once(*args, **dargs)
  File "/home/autotest/autotest-devel/client/tests/virt/virt.py", line 155, in run_once
    run_func(self, params, env)
  File "/home/autotest/autotest-devel/client/tests/virt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/timedrift.py", line 117, in run
    time_format)
  File "/home/autotest/autotest-devel/client/tests/virt/virttest/utils_test/__init__.py", line 307, in get_time
    guest_time = time.mktime(time.strptime(reo, time_format)) + num
  File "/usr/lib64/python2.6/_strptime.py", line 454, in _strptime_time
    return _strptime(data_string, format)[0]
  File "/usr/lib64/python2.6/_strptime.py", line 325, in _strptime
    (data_string, format))
ValueError: time data '0' does not match format '%m/%d/%Y %H:%M:%S'

**Comment by PandaWei** [comment]
OK, ACK

**Comment by PandaWei** [comment]
thanks @spcui and @suqinhuang 
merge now

**Pull request merged**